meson.build

```diff
@@ -0,0 +1,46 @@
+#
+# See LICENSE for more information about licensing
+#  Copyright 2023
+#
+# Author: Diego Arturo Avila Torres <dandida95@gmail.com>
+#         Luis G. Leon Vega <luis.leon@ieee.org>
+#
+#
+
+project(
+  'cynq',
+  'cpp',
+  version : '0.1.0',
+  default_options : ['warning_level=3',
+                     'cpp_std=c++17',
+                     'cpp_args=-Werror'])
+
+add_global_arguments(
+  '-Wno-ignored-qualifiers',
+  language : 'cpp'
+)
+
+#############################
+## Dependencies definition ##
+#############################
+
+
+# options
+
+build_tests = get_option('build_tests')
+
+# directories
+projectinc = include_directories('.',
+'src')
```

Comment:
Headers should be in a directory called "include". Source is only for source code (aka cpp files). So, ideally, this line should be:

```
projectinc = include_directories('include')
```